### PR TITLE
Fix: Diagram component using hardcoded namespace

### LINKED ISSUE #5744: Diagram component using hardcoded default namespace
[Here](https://github.com/backstage/backstage/blob/master/plugins/catalog/src/components/SystemDiagramCard/SystemDiagramCard.tsx#L160) you can see that the diagram component is using `default` namespace hardcoded.

## Expected Behavior

Go to the diagrams tab and see the system and the component linked together.

## Current Behavior

Go to the diagrams tab and see only the system itself.

## Possible Solution

Change this line with something else:
https://github.com/backstage/backstage/blob/master/plugins/catalog/src/components/SystemDiagramCard/SystemDiagramCard.tsx#L160

## Steps to Reproduce

1. Create a system on namespace `my-team` and with name `my-service`
2. Create a component and link to the system with `system: my-team/my-service`
3. Open the system page and click on Diagrams tab

## Context

I was trying to link a component to a system in a specific namespace.

### PR DESCRIPTION
**Pull request by rbrishabh**
Fixes #5744 

Hi @Rugvip! Does this fix work? Thanks!!

### CODE REVIEW

File: plugins/catalog/src/components/SystemDiagramCard/SystemDiagramCard.tsx [hunk 1]
```diff
@@ -157,7 +158,7 @@ export function SystemDiagramCard() {
         kind: ['Component', 'API', 'Resource', 'System', 'Domain'],
         'spec.system': [
           currentSystemName,
-          `${ENTITY_DEFAULT_NAMESPACE}/${currentSystemName}`,
+          `${ref.namespace || 'Current Namespace'}/${currentSystemName}`,
```
Comment (Rugvip) [inline plugins/catalog/src/components/SystemDiagramCard/SystemDiagramCard.tsx:161]: We should be able to use `entity.metadata.namespace` here instead, and then fall back to the default namespace instead.

**Review by Rugvip — COMMENTED** [review]
Thanks!

File: plugins/catalog/src/components/SystemDiagramCard/SystemDiagramCard.tsx (hunk 1 above)
Comment (rbrishabh) [reply plugins/catalog/src/components/SystemDiagramCard/SystemDiagramCard.tsx:161]: i got it! i will make the change

**Comment by rbrishabh** [comment]
Hi @Rugvip! how does this look now? Thanks!

**Review by Rugvip — COMMENTED** [review]
Looks 👌 

Now to be able to ship it we need to add a changeset, or it won't end up being released. Use `yarn changeset` to create one. This'd be a `patch` change since it's a non-breaking bug fix.

**Comment by rbrishabh** [comment]
Hi @Rugvip! done! thanks for your prompt reviews!

File: plugins/catalog/src/components/SystemDiagramCard/SystemDiagramCard.tsx [hunk 2]
```diff
@@ -157,7 +157,7 @@ export function SystemDiagramCard() {
         kind: ['Component', 'API', 'Resource', 'System', 'Domain'],
         'spec.system': [
           currentSystemName,
-          `${ENTITY_DEFAULT_NAMESPACE}/${currentSystemName}`,
+          `${entity.metadata.namespace || ENTITY_DEFAULT_NAMESPACE}/${currentSystemName}`,
```
Comment (Rugvip) [inline plugins/catalog/src/components/SystemDiagramCard/SystemDiagramCard.tsx:160]: Needs formatting, either through editor or `yarn prettier --write plugins/catalog/src/components/SystemDiagramCard/SystemDiagramCard.tsx`

File: .changeset/thick-donkeys-fold.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-catalog': patch
+---
+
+Fix for Diagram component using hardcoded namespace
```
Comment (Rugvip) [inline .changeset/thick-donkeys-fold.md:5]:
```suggestion
Fix for Diagram component using hard coded namespace.
```

**Review by Rugvip — COMMENTED** [review]
Ah, CI uncovered some things

**Comment by rbrishabh** [comment]
whoops! getting right to it!

**Pull request merged**